code/modules/surgery/organs/eyes.dm

```diff
@@ -140,3 +139,181 @@
+	current_color_string = newcolor
+	eye_color = RGB2EYECOLORSTRING(current_color_string)
+	sync_light_effects()
+	cycle_mob_overlay()
+	if(owner)
+		owner.regenerate_icons()
+
+/obj/item/organ/eyes/robotic/glow/proc/cycle_mob_overlay()
+	remove_mob_overlay()
+	mob_overlay.color = current_color_string
+	add_mob_overlay()
+
+/obj/item/organ/eyes/robotic/glow/proc/add_mob_overlay()
+	if(owner)
+		owner.add_overlay(mob_overlay)
+
+/obj/item/organ/eyes/robotic/glow/proc/remove_mob_overlay()
+	if(owner)
+		owner.cut_overlay(mob_overlay)
+
+/obj/item/organ/eyes/robotic/glow/emp_act()
+	if(active)
+		deactivate(silent = TRUE)
+
+/obj/item/organ/eyes/robotic/glow/on_mob_move()
+	if(active)
+		update_visuals()
+
+/obj/item/organ/eyes/robotic/glow/proc/activate(silent = FALSE)
+	start_visuals()
+	if(!silent)
+		to_chat(owner, "<span class='warning'>Your [src] clicks and makes a whining noise, before shooting out a beam of light!</span>")
+	active = TRUE
+	cycle_mob_overlay()
+
+/obj/item/organ/eyes/robotic/glow/proc/deactivate(silent = FALSE)
+	clear_visuals()
+	if(!silent)
+		to_chat(owner, "<span class='warning'>Your [src] shuts off!</span>")
+	active = FALSE
+	remove_mob_overlay()
+
+/obj/item/organ/eyes/robotic/glow/proc/update_visuals()
+	if((eye_lighting.len < light_beam_distance) || !on_mob)
+		regenerate_light_effects()
+	var/turf/scanfrom = get_turf(owner)
+	var/scandir = owner.dir
+	if(!istype(scanfrom))
+		clear_visuals()
+	var/turf/scanning = scanfrom
+	var/stop = FALSE
+	on_mob.forceMove(scanfrom)
+	for(var/i in 1 to light_beam_distance)
+		scanning = get_step(scanning, scandir)
+		if(scanning.opacity || scanning.has_opaque_atom)
+			stop = TRUE
+		var/obj/effect/abstract/eye_lighting/L = eye_lighting[i]
+		if(stop)
+			L.forceMove(src)
+		else
+			L.forceMove(scanning)
+
+/obj/item/organ/eyes/robotic/glow/proc/clear_visuals(delete_everything = FALSE)
+	if(delete_everything)
+		QDEL_LIST(eye_lighting)
+		qdel(on_mob)
+	else
+		for(var/i in eye_lighting)
+			var/obj/effect/abstract/eye_lighting/L = i
+			L.forceMove(src)
+		if(on_mob)
+			on_mob.forceMove(src)
+
+/obj/item/organ/eyes/robotic/glow/proc/start_visuals()
+	if(!initialized)
+		return
+	if((eye_lighting.len < light_beam_distance) || !on_mob)
+		regenerate_light_effects()
+	sync_light_effects()
+	update_visuals()
+
+/obj/item/organ/eyes/robotic/glow/proc/set_distance(dist)
+	light_beam_distance = dist
+	regenerate_light_effects()
+
+/obj/item/organ/eyes/robotic/glow/proc/regenerate_light_effects()
+	QDEL_LIST(eye_lighting)
+	eye_lighting = list()
+	on_mob = new(src)
+	for(var/i in 1 to light_beam_distance)
+		eye_lighting += new /obj/effect/abstract/eye_lighting(src)
+	sync_light_effects()
+
+/obj/item/organ/eyes/robotic/glow/proc/sync_light_effects()
+	for(var/I in eye_lighting)
+		var/obj/effect/abstract/eye_lighting/L = I
+		L.light_color = current_color_string
+		L.set_light(light_object_range)
+		L.light_power = light_object_power
```

Comment:
You can do all of these 3 lines with set_light

`/atom/proc/set_light(var/l_range, var/l_power, var/l_color = NONSENSICAL_VALUE)`